src/main/java/io/github/oliviercailloux/y2018/apartments/valuefunction/profile/Profile.java

```diff
@@ -18,20 +23,10 @@
   private LinearAVF linearAvf;
 
   /**
-   * The 10 next arguments gives the Range of apartment characteristic subjective value weight in
-   * the calculation of the Apartment total subjective value
+   * rangeMap is a map of criterion containing the ranges of all the different range for these
+   * criterion
    */
-  private Range<Double> floorAreaWeightRange;
-
-  private Range<Double> nbBedroomsWeightRange;
-  private Range<Double> nbSleepingWeightRange;
-  private Range<Double> nbBathroomsWeightRange;
-  private Range<Double> terraceWeightRange;
-  private Range<Double> floorAreaTerraceWeightRange;
-  private Range<Double> wifiSubjectiveValueWeightRange;
-  private Range<Double> pricePerNightSubjectiveValueWeightRange;
-  private Range<Double> nbMinNightSubjectiveValueWeightRange;
-  private Range<Double> teleSubjectiveValueWeightRange;
+  private EnumMap<Criterion, Range<Double>> rangeMap;
```

Comment:
`ranges` OPT.

Reply:
Fix in c509ce1eef866b504b08cf8ad8887c137837f441